app/core/pokemon-entity.ts

```diff
@@ -591,11 +591,45 @@ export class PokemonEntity extends Schema implements IPokemonEntity {
   }
 
   addItem(item: Item, permanent = false) {
+    const type = SynergyGivenByItem[item]
+    if (type && this.types.has(type)) {
+      return // prevent adding synergy items for existing types
```

Comment:
about that, we're going to make some changes

![image](https://github.com/user-attachments/assets/b09e2382-07b9-4843-a683-349bb65ff1f5)


for context, the initial prevention for synergy stones was added at a later stage to avoid a footgun for the players. But artificial items can be moved at any time and have other bonuses that just their synergy.

Reply:
allowing duplicate types means that if the item is removed, synergy effects can be removed even though they shouldn't since they still have that type. I couldn't find an easy way to check for this, so I think it should be addressed in another PR.